opencart 1.5.6.4/upload/catalog/controller/smailyforopencart/rss.php

```diff
@@ -0,0 +1,98 @@
+            $this->data['store_url'] = $this->config->get('config_ssl');
+        } else {
+            $this->data['store_url'] = $this->config->get('config_url');
+        }
+        // Build date.
+        $this->data['last_build_date'] = date('D, d M Y H:i:s');
+        // Currency symbol.
+        $this->data['currency'] = $this->session->data['currency'];
+        // Filter for query.
+        $filter = array(
+            'sort' => 'p.date_added',
+            'order' => 'DESC',
+            'start' => 0,
+            'limit' => $limit
+        );
+        // Add category if set.
+        if ($category) {
+            $filter['filter_category_id'] = $this->getCategoryIdByName($category);
+            $filter['filter_sub_caegory'] = true;
+        }
+        // Get items based on filer.
+        $products = $this->model_catalog_product->getProducts($filter);
+        // Items array for template.
+        $items = [];
+        foreach ($products as $product) {
+            $item = [];
+            // Title.
+            $item['title'] = $product['name'];
+            // Link, guid
+            $item['link'] = $this->url->link('product/product', 'product_id=' . $product['product_id']);
+            // Created date.
+            $item['pubDate'] = $product['date_available'];
+            // Description.
+            $item['description'] = $product['description'];
+            // Enclosure.
+            $item['enclosure'] = $this->model_tool_image->resize($product['image'], 300, 300);
+            // Price.
+            $item['price'] = round($product['price'], 2);
+            // Check if product is on sale.
+            if (!empty($product['special']) && $product['special'] > 0) {
+                $item['price'] = round($product['special'], 2);
+                $item['old_price'] = round($product['price'], 2);
+                $item['discount'] = round(100 - ($product['special'] / $product['price'] * 100), 2);
+            }
+            $items[] = $item;
+        }
+        $this->data['items'] = $items;
+        // Return output.
+        $this->template = 'default/template/smailyforopencart/rss.tpl';
+        $this->response->setOutput($this->render());
+    }
+
+    /**
+     * Returns category ID based on category name. Name must be url-encoded.
+     * When no category found with that name, returns empty string.
+     *
+     * @param string $name Category name
+     * @return string $id  Category id.
+     */
+    public function getCategoryIdByName($name) {
+        $this->load->model('catalog/category');
+        $id = '';
+        // All categories.
+        $categories = $this->model_catalog_category->getCategories();
+        foreach ($categories as $category) {
+            if ($category['name'] == $name) {
+                // Get id if name matches.
+                $id = $category['category_id'];
```

Comment:
Add `break` just after this line to stop loop from iterating through rest of the categories when match has been found.

Think of it like this - if you have 200 categories, and the one you are looking for is the first one, then there is no point in looping through the rest of the categories.

This can be also achieved with out using a separate variable:
```php
public function getCategoryIdByName($name) {
    $this->load->model('catalog/category');
    $categories = $this->model_catalog_category->getCategories();
    foreach ($categories as $category) {
        if ($category['name'] == $name) {
            return $category['category_id'];
        }
    }

    // Return empty string, if no matching category is found.
    return '';
}
```